File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if(nums.length <2) return nums.length;
        let counter, maxCounter = 0;
        let set = new Set(nums);
        set.forEach((value) => {
            let item = value;
            counter = 1;
            if(!set.has(value - 1)){
                while(set.has(++item)){
                    counter++;
                }
                maxCounter = Math.max(maxCounter, counter);
            }
        })
        return maxCounter;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = nums[0];
        int local = 0;
        for (int num : nums) {
            local = Math.max(local+num, num);
            max = Math.max(max, local);
        }
        return max;
    }
}
